### PR TITLE
Fix Python 3.13 CICD issues by updating packages

### PR DESCRIPTION
**Pull request by tyler-romero**
Pillow 11 is the first version that supports Python 3.13, using pillow 9 broke test-comprehensive for the Python 3.13 part of the matrix.

Confirmation it works: https://github.com/groundlight/python-sdk/actions/runs/12150031217/job/33882229025?pr=283

### CODE REVIEW

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -119,7 +119,8 @@ jobs:
     if: >-
       github.ref == 'refs/heads/main' ||
       startsWith(github.ref, 'refs/tags/v') ||
-      endsWith(github.ref, '-fulltest')
+      endsWith(github.ref, '-fulltest') ||
+      endsWith(github.ref, '-from-tests-temp')
```
Comment (tyler-romero) [inline .github/workflows/cicd.yaml:123]:
```suggestion
      endsWith(github.ref, '-fulltest')
```
TODO commit this after comprehensive tests pass

File: pyproject.toml [hunk 2]
```diff
@@ -15,8 +15,8 @@ version = "0.19.0"
 # For certifi, use ">=" instead of "^" since it upgrades its "major version" every year, not really following semver
 certifi = ">=2023.7.22"
 frozendict = "^2.3.2"
-pillow = ">=9.0.0" # TODO: We may want to mark pillow (and numpy) as extra (https://python-poetry.org/docs/master/pyproject#extras)
-pydantic = ">=2.0,<3.0.0"
+pillow = ">=11.0.0" # TODO: We may want to mark pillow (and numpy) as extra (https://python-poetry.org/docs/master/pyproject#extras)
```
Comment (tyler-romero) [inline pyproject.toml:18]: pillow 11 is the first version that has support for python 3.13

File: pyproject.toml [hunk 3]
```diff
@@ -42,14 +42,14 @@ types-requests = "^2.28.11.17"
 
 [tool.poetry.group.sphinx-deps.dependencies]
 # These are extra / stricter dependencies required to build the API reference docs
-Sphinx = {version = "7.2.6", python = ">=3.9.0,<4.0"}
-autodoc-pydantic = {version = "2.0.1", python = ">=3.9.0,<4.0"}
-pillow = "^9.0.0"
-pydantic = "^2.0"
-python = ">=3.9.0,<4.0"
+Sphinx = {version = "^7.2.6", python = ">=3.9,<4.0"}
```
Comment (tyler-romero) [inline pyproject.toml:45]: these are all no-ops but use ppetry formatting in the standard way

**Review by brandon-groundlight — APPROVED** [review]
LGTM!

**Pull request merged**